### PR TITLE
Implement User Query and Additional Styling Changes

### PR DESCRIPTION
**Pull request by LuckyHariharan**
Main changes in this PR:

- Wrote and implemented FETCH_USER_PROFILE_CARD query to pull enrolled students and their profile info (name, badges, created_at)
- Hovering over a users card adds a border, and each card contains a link that points to skillify.ca/profile
dashboard

- Users with null as their avatar have the skillify logo as their default profile pic. 

Moving Forward Changes Required:

- Some users have a dead link for their avatar and may need to be set to null in Hasura
- Create input field for current focus and next goal
- Further styling changes to mimic figma mockup
image

### CODE REVIEW

**Review by iamnotreddy — CHANGES_REQUESTED** [review]
a few comments / suggestions but overall great work!

File: pages/coachingdashboard.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+export type Users = {
+  id: string;
+  name: string;
+  profile_image: string;
+  badges_earned: number;
+  created_at: string;
+};
+
+export type ListofUsers = {
```
[the remark this answers is not in the document]
Comment (LuckyHariharan) [reply pages/coachingdashboard.tsx:14]: Sounds good!

File: pages/coachingdashboard.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+          {data?.users.map((it) => (
+            <div className="rounded-md">
+              <ProfileDetailCard
+                avatar={
+                  it.profile_image == null
+                    ? "../../images/logo.png"
+                    : it.profile_image
+                }
+                name={it.name}
+                joinDate={it.created_at.substring(0, 10)}
```
[the remark this answers is not in the document]
Comment (LuckyHariharan) [reply pages/coachingdashboard.tsx:34]: Agreed, used format(new date ...) instead

File: components/coding/studentPortal/ProfileDetailCard.tsx [hunk 3]
```diff
@@ -20,16 +20,51 @@ function ProfileDetailCard({
...
-        <p>next goal: {nextGoal}</p>
-      </div>
+      <a href="profile">
+        <div className="flex flex-col grow items-center gap-4  hover:border-2 hover:rounded-md">
+          <img
+            className="rounded-full h-full border-2 w-20 flex "
+            src={avatar}
+          />
+          <p className=" font-bold "> {name}</p>
+          <div className="flex flex-row">
```
[the remark this answers is not in the document]
Comment (LuckyHariharan) [reply components/coding/studentPortal/ProfileDetailCard.tsx:30]: Awesome!

**Review by iamnotreddy — APPROVED** [review]
approved but remove the extra coaching dashboard file

**Pull request merged**